### DISCUSSION TITLE
How to send and receive pull requests

### DISCUSSION THREAD

**Discussion by dr-nyt** [community/community#22262] [New to GitHub]
<p>Hey there,</p>
<p>I’m new to Github and I’m very confused about how my friend can send change suggestions to my repository, so I can review them and add them to my main branch. Adding contributors was an option, but as far as I know, that gives full access to the repository.</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/dr-nyt">@dr-nyt</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>There are 2 main ways to contribute actively with code to a repository one of this is to be a contributor and the second one is to fork a repository.</p>
<p>The way you take is to add a contributor to your repostiory so your friend can simply do a local clone (in his pc) and make some changes. The best way to follow this choice without making to many conflicts is to create branches for every user and feature that you want to add in order to create pull request after that branch will be pushed to remote.</p>
<p>If you need a help you can try follow the official workflow guide of github -&gt; <a href="https://guides.github.com/introduction/flow/" rel="nofollow noopener">here</a>.</p>
<p>For video guide you can use the youtube videos -&gt; <a href="https://www.youtube.com/githubguides" rel="nofollow noopener">here</a>.</p>
<p>For a workflow more complex and advance -&gt; <a href="https://nvie.com/posts/a-successful-git-branching-model/" rel="nofollow noopener">here</a>.</p>
<p>I hope I have solved your doubts :slightly_smiling_face:</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by dr-nyt** [comment]
<p>Thanks. Appreciate the help :slight_smile:</p>
